core/arch/arm/plat-k3/drivers/ti_crypto.c

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Texas Instruments Crypto Operations
+ *
+ * Copyright (C) 2025 Texas Instruments Incorporated - https://www.ti.com/
+ *	Suhaas Joshi <s-joshi@ti.com>
+ */
+
+#include <platform_config.h>
+#include <drivers/ti_sci.h>
+#include "ti_crypto.h"
+
+TEE_Result ti_crypto_init_rng_fwl(uint16_t fwl_id, uint16_t fwl_region)
+
+{
+	uint16_t rng_region = RNG_TI_SCI_FW_RGN_ID;
+	uint8_t owner_index = OPTEE_HOST_ID;
+	uint8_t owner_privid = 0;
+	uint16_t owner_permission_bits = 0;
+	uint32_t control = 0;
+	uint32_t permissions[FWL_MAX_PRIVID_SLOTS] = { };
+	uint32_t num_perm = 0;
+	uint64_t start_address = 0;
+	uint64_t end_address = 0;
+	TEE_Result ret = TEE_SUCCESS;
+
+	/* Try to claim background firewall region for ourselves */
+	ret = ti_sci_change_fwl_owner(fwl_id, fwl_region, owner_index,
+				      &owner_privid, &owner_permission_bits);
+	if (ret) {
+#if !defined(PLATFORM_FLAVOR_am62lx)
```

Comment:
This wasn't in the original. Do only the refactor in its own commit, all other changes go in follow up commits. That makes it easier to see what you are changing vs just moving around.